### PR TITLE
[DOCS] Adding Meltano implementation pattern to docs

### PR DESCRIPTION
**Pull request by pnadolny13**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- adding new docs section for using Great Expectations with Meltano
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Comment by pnadolny13** [comment]
@cla-bot check

File: docs/deployment_patterns/how_to_use_great_expectations_with_meltano.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+       Expectations for the source data in the source system.
+       Does my Postgres DB (or any source) data meet expectations before I replicate it to my warehouse?
+       Are their source data problems I should be aware of?
+
+1. **Profiling For Migration**
+
+       As part of a migration between warehouses, profiling can give confidence that the data in the new warehouse meets expectations by matching the profile of the original warehouse.
+       Am I confident that my new warehouse has all my data before switching over?
+
+1. **Transformation Between Steps**
```
Comment (austiezr) [inline docs/deployment_patterns/how_to_use_great_expectations_with_meltano.md:126]: Should these numbered cases be increasing?

**Review by austiezr — APPROVED** [review]
Minor note; not gating. Love to see this!

File: docs/deployment_patterns/how_to_use_great_expectations_with_meltano.md (hunk 1 above)
Comment (pnadolny13) [reply docs/deployment_patterns/how_to_use_great_expectations_with_meltano.md:126]: @austiezr thanks for the review! They actually do render as increasing. I find it easier to maintain if you need to reorder or add another item to the list if you dont have to update all explicitly. Happy to change it though if it breaks conventions!

Comment (austiezr) [reply docs/deployment_patterns/how_to_use_great_expectations_with_meltano.md:126]: If it works, it works! Thank you @pnadolny13 !

**Review by donaldheppner — APPROVED** [review]

**Pull request merged**